### ISSUE TITLE
Flatpak enter fails for Atom

### ISSUE THREAD

**Issue by rugk** [flatpak/flatpak#2767]
## Linux distribution and version
Fedora 29

## Description of the problem

Flatpak enter does not work, similar to https://github.com/flatpak/flatpak/issues/855

It only does not work for the Atom flatpak from flathub…

## Steps to reproduce
It still does not work in my case… strangely…

Note that `flatpak ps`works with sudo, so it does not seem to be the problem of sudo…

(Fedora 29)

```sh
$ sudo -E flatpak ps                   
Instance   PID   Application  Runtime
2725877937 31125 io.atom.Atom org.freedesktop.Sdk
$ sudo -E flatpak enter 31125 /bin/bash
error: No such pid 31125
```

## Info

```sh
flatpak info --show-permissions io.atom.Atom            
[Context]
shared=network;ipc;
sockets=x11;pulseaudio;ssh-auth;
devices=all;
filesystems=xdg-config/kdeglobals:ro;host;

[Session Bus Policy]
com.canonical.AppMenu.Registrar.*=talk
com.canonical.AppMenu.Registrar=talk
org.gtk.vfs.*=talk
org.freedesktop.secrets=talk
org.gtk.vfs=talk
org.freedesktop.Flatpak=talk
org.freedesktop.Notifications=talk

[Environment]
ELECTRON_TRASH=this-is-a-workaround
```

**Comment by kirbyfan64** [comment]
Does using the instance ID instead of the PID work instead?

**Comment by rugk** [comment]
I've tried all of them. And no, I've used the short number, i.e. PID. I know it is the correct one, as it, as said, works when I use the QT flatpak demo.

**Comment by kirbyfan64** [comment]
For the purposes of testing, can you try the instance ID (the longer one)? They should both work.

**Comment by rugk** [comment]
As I said, I've tried everything.

Could you try to reproduce it with this flatpak? Maybe it is really only specific to this flatpak.

**Comment by kirbyfan64** [comment]
I can definitely reproduce this. Bizarrely enough, the PID recorded in `$XDG_RUNTIME_DIR/.flatpak/instance/pid` is *dead*, and the child PID recorded in `$XDG_RUNTIME_DIR/.flatpak/instance/bwrapinfo.json` is the bwrap process.

Atom seems to go through several intermediate processes before settling on its main one, maybe that could be related to it?

**Comment by rugk** [comment]
Okay, so also reported on https://github.com/flathub/io.atom.Atom/issues/44 to the maintainer of the specific flatpak. Let's see what could be the error here.

**Comment by rugk** [comment]
Workaround BTW: `flatpak run --command=bash io.atom.Atom`

**Comment by clopez** [comment]
I think ```flatpak enter``` is currently broken. See #3059

**Comment by matthiasclasen** [comment]
try using sudo -E

**Comment by rugk** [comment]
Please look at the "Steps to reproduce", there I already wrote the exact commends. 

I just reproduced:
```
$ sudo -E flatpak ps                   

Note that the directory 

'/root/.local/share/flatpak/exports/share'

is not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

Instance   PID   Application             Runtime
1037171777 23039 io.atom.Atom            org.freedesktop.Sdk
1667152164 21084 org.keepassxc.KeePassXC org.kde.Platform
3036473854 20782 org.keepassxc.KeePassXC org.kde.Platform

$ sudo -E flatpak enter 23039 /bin/bash

Note that the directory 

'/root/.local/share/flatpak/exports/share'

is not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

Fehler: Prozesskennung 23039 nicht vorhanden
```

Fedora 31 Silverblue

**Note it only does not work with Atom** in my case, other flatpaks worked. (I tried KeePassXC that is also started)

**Comment by matthiasclasen** [comment]
I believe master should work better.

**Comment by Yajo** [comment]
I'm having the same problem with code:

<pre>yajo@yajolap <font color="#4E9A06">~</font>&gt; <font color="#005FD7">flatpak</font> <font color="#00AFFF">ps</font>
<b>Instance   PID   Application               Runtime</b>
3761682687 28730 com.visualstudio.code.oss org.freedesktop.Sdk
yajo@yajolap <font color="#4E9A06">~</font>&gt; <font color="#005FD7">flatpak</font> <font color="#00AFFF">enter</font> <font color="#00AFFF">3761682687</font> <font color="#00AFFF">code-oss</font> <font color="#00AFFF">--wait</font>
Not running as root, may be unable to enter namespace
<font color="#CC0000"><b>error: </b></font>No such pid 3761682687
yajo@yajolap <font color="#4E9A06">~</font>&gt; <font color="#005FD7">flatpak</font> <font color="#00AFFF">enter</font> <font color="#00AFFF">28730</font> <font color="#00AFFF">code-oss</font> <font color="#00AFFF">--wait</font>
Not running as root, may be unable to enter namespace
<font color="#CC0000"><b>error: </b></font>No such pid 28730
yajo@yajolap <font color="#4E9A06">~</font>&gt; <font color="#005FD7">sudo</font> <font color="#00AFFF">flatpak</font> <font color="#00AFFF">enter</font> <font color="#00AFFF">3761682687</font> <font color="#00AFFF">code-oss</font> <font color="#00AFFF">--wait</font>
<font color="#CC0000"><b>error: </b></font>3761682687 is neither a pid nor an application or instance ID
yajo@yajolap <font color="#4E9A06">~</font>&gt; <font color="#005FD7">sudo</font> <font color="#00AFFF">flatpak</font> <font color="#00AFFF">enter</font> <font color="#00AFFF">28730</font> <font color="#00AFFF">code-oss</font> <font color="#00AFFF">--wait</font>
<font color="#CC0000"><b>error: </b></font>28730 is neither a pid nor an application or instance ID
yajo@yajolap <font color="#4E9A06">~</font>&gt; <font color="#005FD7">sudo</font> <font color="#00AFFF">-E</font> <font color="#00AFFF">flatpak</font> <font color="#00AFFF">enter</font> <font color="#00AFFF">3761682687</font> <font color="#00AFFF">code-oss</font> <font color="#00AFFF">--wait</font>

Note that the directory 

&apos;/root/.local/share/flatpak/exports/share&apos;

is not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

<font color="#CC0000"><b>error: </b></font>No such pid 3761682687
yajo@yajolap <font color="#4E9A06">~</font>&gt; <font color="#005FD7">sudo</font> <font color="#00AFFF">-E</font> <font color="#00AFFF">flatpak</font> <font color="#00AFFF">enter</font> <font color="#00AFFF">28730</font> <font color="#00AFFF">code-oss</font> <font color="#00AFFF">--wait</font>

Note that the directory 

&apos;/root/.local/share/flatpak/exports/share&apos;

is not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

<font color="#CC0000"><b>error: </b></font>No such pid 28730
</pre>

Basically, no way to `flatpak exec`.

Using F31 Silverblue.

**Comment by alexlarsson** [comment]
Flatpak enter is a bit lame in earlier versions. Post 1.5.1 it is a lot better, and doesn't require sudo.

**Comment by Mikenux** [comment]
Work here on Fedora 40, flatpak 1.15.10.

@rugk, @yajo?

**Comment by rugk** [comment]
Atom is discontinued now, AFAIK. So I don't use it anymore…

**Comment by Mikenux** [comment]
Thanks. Let's wait for the other person's reply then.

**Comment by yajo** [comment]
Same here, sorry
